src/pagination/index.tsx

```diff
@@ -53,20 +71,27 @@ interface PaginationCache {
 	currentPage: number;
 	pageSize: number;
 	pageSizes: number[];
-	pageSizesData: ListOption[];
+	pageSizesResource: { resource: () => Resource; data: ListOption[] };
```

Comment:
I think this should stay as `ListOption[]`, there's no need to expose the resource usage to the consumer of this pagination widget

Reply:
The intention was to only create the source when necessary (ie, when the `pageSizeData` had changed).

Reply:
I think I've incorporated correct resource usage, but I'm still storing the resource in the cache. I'll let you resolve this conversation if you agree.